src/main/java/com/exactpro/th2/inframgr/Config.java

```diff
@@ -127,17 +128,9 @@ public void setKubernetes(K8sConfig kubernetes) {
 
     private Config() {}
 
-    public static Config getInstance() throws IOException {
-        if (instance == null) {
-            synchronized (Config.class) {
-                if (instance == null) {
-                    Path file = CONFIG_DIR.resolve(CONFIG_FILE);
-                    instance = readConfiguration(file);
-                }
-            }
-        }
-
-        return instance;
+    public static Config instance() throws IOException {
```

Comment:
This still sounds like a singleton method. Probably we should call it `createInstance` or something like that